feat(ai-gateway): AI Gateway 2.0 entities (#5057)
Kong/developer.konghq.com: app/_ai_gateway_entities/ai-gateway.md

---

> title: AI Gateways

Reviewer: No links to this on the entities page yet

Reviewer 2: fixed

content_type: reference
entities:
  - ai-gateway
products:
  - ai-gateway
min_version:
  ai-gateway: '2.0.0'
breadcrumbs:
  - /ai-gateway/
  - /ai-gateway/entities/
description: The top-level {{site.ai_gateway}} entity that owns Models, Providers, Policies, Agents, MCP Servers, and other AI-specific entities.
schema:
  api: konnect/ai-gateway
  path: /schemas/AIGateway
works_on:
  - konnect
related_resources:
  - text: About {{site.ai_gateway}}
    url: /ai-gateway/
  - text: "{{site.ai_gateway}} entities"
    url: /ai-gateway/entities/
  - text: Model entity
    url: /ai-gateway/entities/model/
  - text: Provider entity
    url: /ai-gateway/entities/provider/
  - text: Policy entity
    url: /ai-gateway/entities/policy/
  - text: Data Plane Certificate entity
    url: /ai-gateway/entities/data-plane-certificate/

> faqs:

Reviewer: These FAQs are great

Reviewer 2: Thanks!

1. **Configuration endpoint** (`endpoints.configuration`) — the URL data plane nodes use to receive their configuration from the control plane.
1. **Telemetry endpoint** (`endpoints.telemetry`) — the URL data plane nodes use to ship analytics and runtime telemetry back to {{site.konnect_short_name}}.

Both endpoints are read-only, assigned at creation time, and stable for the lifetime of the AI Gateway. Data plane nodes need both URLs, along with a [Data Plane Certificate](/ai-gateway/entities/data-plane-certificate/), to register with the AI Gateway.

## Configuration hash

`config_hash` is a read-only field that {{site.ai_gateway}} updates every time anything under the AI Gateway changes — a new Model, an updated Policy, a deleted Provider, and so on. Each data plane node reports back the `config_hash` of the configuration it's running. The two values match when the node is in sync with the control plane.

Use `config_hash` to verify rollout: after a configuration change, watch the node `config_hash` (through [List Nodes](/ai-gateway/entities/data-plane-certificate/) or the {{site.konnect_short_name}} UI) until every node reports the AI Gateway's current value.

## Labels

> `labels` is a free-form `key: value` map for organization. Use it to tag AI Gateways by environment (`env: production`), team ownership, cost center, or any other dimension you filter on. Labels don't affect runtime behavior.

Reviewer: ```suggestion
`labels` are a free-form `key: value` map for organization. Use them to tag AI Gateways by environment (`env: production`), team ownership, cost center, or any other dimension you filter on. Labels don't affect runtime behavior.
```
